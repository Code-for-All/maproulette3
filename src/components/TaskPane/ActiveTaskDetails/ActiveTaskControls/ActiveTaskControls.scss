@import '../../../../variables.scss';

.active-task-controls {
  &__control-block {
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
    margin-top: 10px;
    margin-bottom: 10px;


    button.button.large-and-wide {
      margin-bottom: 10px;

      .control-icon svg {
        height: 20px;
        width: auto;
        margin-right: 10px;
      }
    }
  }

  &__vertical-control-block {
    display: flex;
    flex-direction: column;

    button {
      margin-bottom: 10px;
    }
  }

  &__track-task {
    margin-bottom: 20px;
  }

  &.is-minimized {
    .active-task-controls__control-block {
      flex-direction: column;

      .control-label {
        display: none;
      }
    }

    button {
      margin-bottom: 5px;
    }

    .active-task-controls__icon {
      height: 26px;
      width: auto;
      margin-left: 0px;
      margin-right: 0px;
      fill: $default-icon-color;
    }

    .active-task-controls__task-comment {
      display: none;
    }

    .control-label {
      display: none;
    }
  }
}
