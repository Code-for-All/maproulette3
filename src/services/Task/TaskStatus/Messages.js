import { defineMessages } from 'react-intl'

/**
 * Internationalized messages for use with TaskStatus.
 */
export default defineMessages({
  created: {
    id: "Task.status.created",
    defaultMessage: "Created"
  },
  fixed: {
    id: "Task.status.fixed",
    defaultMessage: "Fixed"
  },
  falsePositive: {
    id: "Task.status.falsePositive",
    defaultMessage: "Not an Issue"
  },
  skipped: {
    id: "Task.status.skipped",
    defaultMessage: "Skipped"
  },
  deleted: {
    id: "Task.status.deleted",
    defaultMessage: "Deleted"
  },
  alreadyFixed: {
    id: "Task.status.alreadyFixed",
    defaultMessage: "Already Fixed"
  },
  tooHard: {
    id: "Task.status.tooHard",
    defaultMessage: "Too Hard"
  },
})
