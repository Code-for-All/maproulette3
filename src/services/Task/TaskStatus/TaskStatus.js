import { map as _map,
         invert as _invert,
         fromPairs as _fromPairs } from 'lodash'
import messages from './Messages'

// These statuses are defined on the server
export const TASK_STATUS_CREATED = 0
export const TASK_STATUS_FIXED = 1
export const TASK_STATUS_FALSE_POSITIVE = 2
export const TASK_STATUS_SKIPPED = 3
export const TASK_STATUS_DELETED = 4
export const TASK_STATUS_ALREADY_FIXED = 5
export const TASK_STATUS_TOO_HARD = 6

export const TaskStatus = Object.freeze({
  created: TASK_STATUS_CREATED,
  fixed: TASK_STATUS_FIXED,
  falsePositive: TASK_STATUS_FALSE_POSITIVE,
  skipped: TASK_STATUS_SKIPPED,
  deleted: TASK_STATUS_DELETED,
  alreadyFixed: TASK_STATUS_ALREADY_FIXED,
  tooHard: TASK_STATUS_TOO_HARD,
})

export const keysByStatus = Object.freeze(_invert(TaskStatus))

/**
 * Returns true if the given status represents completion of
 * the task.
 */
export const isCompleted = function(status) {
  return status === TaskStatus.fixed ||
         status === TaskStatus.falsePositive ||
         status === TaskStatus.alreadyFixed ||
         status === TaskStatus.deleted
}

/**
 * Returns an object mapping difficulty values to raw internationalized
 * messages suitable for use with FormattedMessage or formatMessage.
 */
export const messagesByStatus = _fromPairs(
  _map(messages, (message, key) => [TaskStatus[key], message])
)

/** Returns object containing localized labels  */
export const statusLabels = intl => _fromPairs(
  _map(messages, (message, key) => [key, intl.formatMessage(message)])
)
