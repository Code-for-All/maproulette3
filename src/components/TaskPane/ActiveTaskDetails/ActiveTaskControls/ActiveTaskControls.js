import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { injectIntl } from 'react-intl'
import { get as _get } from 'lodash'
import { isCompleted } from '../../../../services/Task/TaskStatus/TaskStatus'
import WithMapBoundsState from '../../../HOCs/WithMapBounds/WithMapBoundsState'
import WithKeyboardShortcuts from '../../../HOCs/WithKeyboardShortcuts/WithKeyboardShortcuts'
import BusySpinner from '../../../BusySpinner/BusySpinner'
import TaskEditControls from './TaskEditControls/TaskEditControls'
import TaskCompletionControls from './TaskCompletionControls/TaskCompletionControls'
import TaskDoneControls from './TaskDoneControls/TaskDoneControls'

/**
 * ActiveTaskControls renders the appropriate controls for the given
 * active task based on the state of the task and editing workflow.
 *
 * @author [Neil Rotstan](https://github.com/nrotstan)
 */
export class ActiveTaskControls extends Component {
  state = {
    taskBeingCompleted: null,
  }

  setTaskBeingCompleted = (taskId) => {
    this.setState({taskBeingCompleted: taskId})
  }

  render() {
    if (!this.props.task) {
      return null
    }

    const isEditingTask =
      _get(this.props, 'editor.taskId') === this.props.task.id &&
      _get(this.props, 'editor.success') === true

    if (isCompleted(_get(this.props, 'task.status'))) {
      return <TaskDoneControls {...this.props} />
    }
    else if (isEditingTask) {
      // Editor is open, show completion options
      return <TaskCompletionControls setTaskBeingCompleted={this.setTaskBeingCompleted}
                                     {...this.props} />
    }
    else if (_get(this.props, 'editor.taskId') !== this.props.task.id &&
             this.state.taskBeingCompleted === this.props.task.id) {
      // Busy spinner until editor catches up with us
      return <BusySpinner />
    }
    else {
      return <TaskEditControls setTaskBeingCompleted={this.setTaskBeingCompleted}
                               {...this.props} />
    }
  }
}

ActiveTaskControls.propTypes = {
  task: PropTypes.object,
  editor: PropTypes.object,
}

ActiveTaskControls.defaultProps = {
  editor: {},
}

export default WithMapBoundsState(
  WithKeyboardShortcuts(
    injectIntl(ActiveTaskControls)
  )
)
