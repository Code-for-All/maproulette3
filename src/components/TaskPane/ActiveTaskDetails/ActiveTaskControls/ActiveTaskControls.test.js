import React from 'react'
import { omit as _omit, cloneDeep as _cloneDeep } from 'lodash'
import { ActiveTaskControls } from './ActiveTaskControls'
import keyMappings from '../../../../KeyMappings'
import { TaskStatus } from '../../../../services/Task/TaskStatus/TaskStatus'

const propsFixture = {
  task: {
    id: 123,
  },
  editor: {
  },
  mapBounds: {
    task: {
      bounds: [0, 0, 0, 0],
    }
  },
  keyboardShortcutGroups: keyMappings,
  user: {
    id: 357,
    settings: {defaultEditor: 1},
  },
}

let basicProps = null

beforeEach(() => {
  basicProps = _cloneDeep(propsFixture)

  basicProps.editTask = jest.fn()
  basicProps.completeTask = jest.fn()
  basicProps.nextTask = jest.fn()
  basicProps.setTaskBeingCompleted = jest.fn()
  basicProps.activateKeyboardShortcuts = jest.fn()
  basicProps.deactivateKeyboardShortcuts = jest.fn()
  basicProps.closeEditor = jest.fn()
  basicProps.intl = {formatMessage: jest.fn()}
})

test("already-done controls are shown if the task is in a completed status.", () => {
  basicProps.task.status = TaskStatus.fixed
  const wrapper = shallow(
    <ActiveTaskControls {...basicProps} />
  )

  expect(wrapper.find('TaskDoneControls').exists()).toBe(true)
  expect(wrapper.find('TaskEditControls').exists()).toBe(false)
  expect(wrapper.find('TaskCompletionControls').exists()).toBe(false)

  expect(wrapper).toMatchSnapshot()
})

test("completion controls are shown if the user has begun editing the task", () => {
  basicProps.task.status = TaskStatus.created
  basicProps.editor.taskId = basicProps.task.id
  basicProps.editor.success = true

  const wrapper = shallow(
    <ActiveTaskControls {...basicProps} />
  )

  expect(wrapper.find('TaskCompletionControls').exists()).toBe(true)
  expect(wrapper.find('TaskDoneControls').exists()).toBe(false)
  expect(wrapper.find('TaskEditControls').exists()).toBe(false)

  expect(wrapper).toMatchSnapshot()
})

test("edit controls are shown if the user has not yet begun editing the task", () => {
  basicProps.task.status = TaskStatus.created

  const wrapper = shallow(
    <ActiveTaskControls {...basicProps} />
  )

  expect(wrapper.find('TaskEditControls').exists()).toBe(true)
  expect(wrapper.find('TaskCompletionControls').exists()).toBe(false)
  expect(wrapper.find('TaskDoneControls').exists()).toBe(false)

  expect(wrapper).toMatchSnapshot()
})

test("a busy spinner is shown when switching from edit controls to completion controls", () => {
  basicProps.task.status = TaskStatus.created

  const wrapper = shallow(
    <ActiveTaskControls {...basicProps} />
  )
  
  wrapper.instance().setTaskBeingCompleted(basicProps.task.id)
  wrapper.update()

  expect(wrapper.find('BusySpinner').exists()).toBe(true)
  expect(wrapper.find('TaskEditControls').exists()).toBe(false)
  expect(wrapper.find('TaskCompletionControls').exists()).toBe(false)
  expect(wrapper.find('TaskDoneControls').exists()).toBe(false)

  expect(wrapper).toMatchSnapshot()
})
