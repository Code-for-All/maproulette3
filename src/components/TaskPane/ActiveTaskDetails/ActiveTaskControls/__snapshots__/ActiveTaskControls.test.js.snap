// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a busy spinner is shown when switching from edit controls to completion controls 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActiveTaskControls
    activateKeyboardShortcuts={[Function]}
    closeEditor={[Function]}
    completeTask={[Function]}
    deactivateKeyboardShortcuts={[Function]}
    editTask={[Function]}
    editor={Object {}}
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
    keyboardShortcutGroups={
        Object {
            "taskCompletion": Object {
              "cancel": Object {
                "key": "Escape",
                "keyLabel": "ESC",
                "label": "Cancel Editing",
              },
            },
            "taskEditing": Object {
              "editId": Object {
                "key": "e",
                "label": "Edit in Id",
              },
              "editJosm": Object {
                "key": "r",
                "label": "Edit in JOSM",
              },
              "editJosmLayer": Object {
                "key": "t",
                "label": "Edit in new JOSM layer",
              },
              "falsePositive": Object {
                "key": "q",
                "label": "Not an Issue",
              },
              "skip": Object {
                "key": "w",
                "label": "Skip",
              },
            },
          }
    }
    mapBounds={
        Object {
            "task": Object {
              "bounds": Array [
                0,
                0,
                0,
                0,
              ],
            },
          }
    }
    nextTask={[Function]}
    setTaskBeingCompleted={[Function]}
    task={
        Object {
            "id": 123,
            "status": 0,
          }
    }
    user={
        Object {
            "id": 357,
            "settings": Object {
              "defaultEditor": 1,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`already-done controls are shown if the task is in a completed status. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActiveTaskControls
    activateKeyboardShortcuts={[Function]}
    closeEditor={[Function]}
    completeTask={[Function]}
    deactivateKeyboardShortcuts={[Function]}
    editTask={[Function]}
    editor={Object {}}
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
    keyboardShortcutGroups={
        Object {
            "taskCompletion": Object {
              "cancel": Object {
                "key": "Escape",
                "keyLabel": "ESC",
                "label": "Cancel Editing",
              },
            },
            "taskEditing": Object {
              "editId": Object {
                "key": "e",
                "label": "Edit in Id",
              },
              "editJosm": Object {
                "key": "r",
                "label": "Edit in JOSM",
              },
              "editJosmLayer": Object {
                "key": "t",
                "label": "Edit in new JOSM layer",
              },
              "falsePositive": Object {
                "key": "q",
                "label": "Not an Issue",
              },
              "skip": Object {
                "key": "w",
                "label": "Skip",
              },
            },
          }
    }
    mapBounds={
        Object {
            "task": Object {
              "bounds": Array [
                0,
                0,
                0,
                0,
              ],
            },
          }
    }
    nextTask={[Function]}
    setTaskBeingCompleted={[Function]}
    task={
        Object {
            "id": 123,
            "status": 1,
          }
    }
    user={
        Object {
            "id": 357,
            "settings": Object {
              "defaultEditor": 1,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activateKeyboardShortcuts": [Function],
      "closeEditor": [Function],
      "completeTask": [Function],
      "deactivateKeyboardShortcuts": [Function],
      "editTask": [Function],
      "editor": Object {},
      "intl": Object {
        "formatMessage": [Function],
      },
      "keyboardShortcutGroups": Object {
        "taskCompletion": Object {
          "cancel": Object {
            "key": "Escape",
            "keyLabel": "ESC",
            "label": "Cancel Editing",
          },
        },
        "taskEditing": Object {
          "editId": Object {
            "key": "e",
            "label": "Edit in Id",
          },
          "editJosm": Object {
            "key": "r",
            "label": "Edit in JOSM",
          },
          "editJosmLayer": Object {
            "key": "t",
            "label": "Edit in new JOSM layer",
          },
          "falsePositive": Object {
            "key": "q",
            "label": "Not an Issue",
          },
          "skip": Object {
            "key": "w",
            "label": "Skip",
          },
        },
      },
      "mapBounds": Object {
        "task": Object {
          "bounds": Array [
            0,
            0,
            0,
            0,
          ],
        },
      },
      "nextTask": [Function],
      "setTaskBeingCompleted": [Function],
      "task": Object {
        "id": 123,
        "status": 1,
      },
      "user": Object {
        "id": 357,
        "settings": Object {
          "defaultEditor": 1,
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activateKeyboardShortcuts": [Function],
        "closeEditor": [Function],
        "completeTask": [Function],
        "deactivateKeyboardShortcuts": [Function],
        "editTask": [Function],
        "editor": Object {},
        "intl": Object {
          "formatMessage": [Function],
        },
        "keyboardShortcutGroups": Object {
          "taskCompletion": Object {
            "cancel": Object {
              "key": "Escape",
              "keyLabel": "ESC",
              "label": "Cancel Editing",
            },
          },
          "taskEditing": Object {
            "editId": Object {
              "key": "e",
              "label": "Edit in Id",
            },
            "editJosm": Object {
              "key": "r",
              "label": "Edit in JOSM",
            },
            "editJosmLayer": Object {
              "key": "t",
              "label": "Edit in new JOSM layer",
            },
            "falsePositive": Object {
              "key": "q",
              "label": "Not an Issue",
            },
            "skip": Object {
              "key": "w",
              "label": "Skip",
            },
          },
        },
        "mapBounds": Object {
          "task": Object {
            "bounds": Array [
              0,
              0,
              0,
              0,
            ],
          },
        },
        "nextTask": [Function],
        "setTaskBeingCompleted": [Function],
        "task": Object {
          "id": 123,
          "status": 1,
        },
        "user": Object {
          "id": 357,
          "settings": Object {
            "defaultEditor": 1,
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`completion controls are shown if the user has begun editing the task 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActiveTaskControls
    activateKeyboardShortcuts={[Function]}
    closeEditor={[Function]}
    completeTask={[Function]}
    deactivateKeyboardShortcuts={[Function]}
    editTask={[Function]}
    editor={
        Object {
            "success": true,
            "taskId": 123,
          }
    }
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
    keyboardShortcutGroups={
        Object {
            "taskCompletion": Object {
              "cancel": Object {
                "key": "Escape",
                "keyLabel": "ESC",
                "label": "Cancel Editing",
              },
            },
            "taskEditing": Object {
              "editId": Object {
                "key": "e",
                "label": "Edit in Id",
              },
              "editJosm": Object {
                "key": "r",
                "label": "Edit in JOSM",
              },
              "editJosmLayer": Object {
                "key": "t",
                "label": "Edit in new JOSM layer",
              },
              "falsePositive": Object {
                "key": "q",
                "label": "Not an Issue",
              },
              "skip": Object {
                "key": "w",
                "label": "Skip",
              },
            },
          }
    }
    mapBounds={
        Object {
            "task": Object {
              "bounds": Array [
                0,
                0,
                0,
                0,
              ],
            },
          }
    }
    nextTask={[Function]}
    setTaskBeingCompleted={[Function]}
    task={
        Object {
            "id": 123,
            "status": 0,
          }
    }
    user={
        Object {
            "id": 357,
            "settings": Object {
              "defaultEditor": 1,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activateKeyboardShortcuts": [Function],
      "closeEditor": [Function],
      "completeTask": [Function],
      "deactivateKeyboardShortcuts": [Function],
      "editTask": [Function],
      "editor": Object {
        "success": true,
        "taskId": 123,
      },
      "intl": Object {
        "formatMessage": [Function],
      },
      "keyboardShortcutGroups": Object {
        "taskCompletion": Object {
          "cancel": Object {
            "key": "Escape",
            "keyLabel": "ESC",
            "label": "Cancel Editing",
          },
        },
        "taskEditing": Object {
          "editId": Object {
            "key": "e",
            "label": "Edit in Id",
          },
          "editJosm": Object {
            "key": "r",
            "label": "Edit in JOSM",
          },
          "editJosmLayer": Object {
            "key": "t",
            "label": "Edit in new JOSM layer",
          },
          "falsePositive": Object {
            "key": "q",
            "label": "Not an Issue",
          },
          "skip": Object {
            "key": "w",
            "label": "Skip",
          },
        },
      },
      "mapBounds": Object {
        "task": Object {
          "bounds": Array [
            0,
            0,
            0,
            0,
          ],
        },
      },
      "nextTask": [Function],
      "setTaskBeingCompleted": [Function],
      "task": Object {
        "id": 123,
        "status": 0,
      },
      "user": Object {
        "id": 357,
        "settings": Object {
          "defaultEditor": 1,
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activateKeyboardShortcuts": [Function],
        "closeEditor": [Function],
        "completeTask": [Function],
        "deactivateKeyboardShortcuts": [Function],
        "editTask": [Function],
        "editor": Object {
          "success": true,
          "taskId": 123,
        },
        "intl": Object {
          "formatMessage": [Function],
        },
        "keyboardShortcutGroups": Object {
          "taskCompletion": Object {
            "cancel": Object {
              "key": "Escape",
              "keyLabel": "ESC",
              "label": "Cancel Editing",
            },
          },
          "taskEditing": Object {
            "editId": Object {
              "key": "e",
              "label": "Edit in Id",
            },
            "editJosm": Object {
              "key": "r",
              "label": "Edit in JOSM",
            },
            "editJosmLayer": Object {
              "key": "t",
              "label": "Edit in new JOSM layer",
            },
            "falsePositive": Object {
              "key": "q",
              "label": "Not an Issue",
            },
            "skip": Object {
              "key": "w",
              "label": "Skip",
            },
          },
        },
        "mapBounds": Object {
          "task": Object {
            "bounds": Array [
              0,
              0,
              0,
              0,
            ],
          },
        },
        "nextTask": [Function],
        "setTaskBeingCompleted": [Function],
        "task": Object {
          "id": 123,
          "status": 0,
        },
        "user": Object {
          "id": 357,
          "settings": Object {
            "defaultEditor": 1,
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`edit controls are shown if the user has not yet begun editing the task 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActiveTaskControls
    activateKeyboardShortcuts={[Function]}
    closeEditor={[Function]}
    completeTask={[Function]}
    deactivateKeyboardShortcuts={[Function]}
    editTask={[Function]}
    editor={Object {}}
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
    keyboardShortcutGroups={
        Object {
            "taskCompletion": Object {
              "cancel": Object {
                "key": "Escape",
                "keyLabel": "ESC",
                "label": "Cancel Editing",
              },
            },
            "taskEditing": Object {
              "editId": Object {
                "key": "e",
                "label": "Edit in Id",
              },
              "editJosm": Object {
                "key": "r",
                "label": "Edit in JOSM",
              },
              "editJosmLayer": Object {
                "key": "t",
                "label": "Edit in new JOSM layer",
              },
              "falsePositive": Object {
                "key": "q",
                "label": "Not an Issue",
              },
              "skip": Object {
                "key": "w",
                "label": "Skip",
              },
            },
          }
    }
    mapBounds={
        Object {
            "task": Object {
              "bounds": Array [
                0,
                0,
                0,
                0,
              ],
            },
          }
    }
    nextTask={[Function]}
    setTaskBeingCompleted={[Function]}
    task={
        Object {
            "id": 123,
            "status": 0,
          }
    }
    user={
        Object {
            "id": 357,
            "settings": Object {
              "defaultEditor": 1,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activateKeyboardShortcuts": [Function],
      "closeEditor": [Function],
      "completeTask": [Function],
      "deactivateKeyboardShortcuts": [Function],
      "editTask": [Function],
      "editor": Object {},
      "intl": Object {
        "formatMessage": [Function],
      },
      "keyboardShortcutGroups": Object {
        "taskCompletion": Object {
          "cancel": Object {
            "key": "Escape",
            "keyLabel": "ESC",
            "label": "Cancel Editing",
          },
        },
        "taskEditing": Object {
          "editId": Object {
            "key": "e",
            "label": "Edit in Id",
          },
          "editJosm": Object {
            "key": "r",
            "label": "Edit in JOSM",
          },
          "editJosmLayer": Object {
            "key": "t",
            "label": "Edit in new JOSM layer",
          },
          "falsePositive": Object {
            "key": "q",
            "label": "Not an Issue",
          },
          "skip": Object {
            "key": "w",
            "label": "Skip",
          },
        },
      },
      "mapBounds": Object {
        "task": Object {
          "bounds": Array [
            0,
            0,
            0,
            0,
          ],
        },
      },
      "nextTask": [Function],
      "setTaskBeingCompleted": [Function],
      "task": Object {
        "id": 123,
        "status": 0,
      },
      "user": Object {
        "id": 357,
        "settings": Object {
          "defaultEditor": 1,
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activateKeyboardShortcuts": [Function],
        "closeEditor": [Function],
        "completeTask": [Function],
        "deactivateKeyboardShortcuts": [Function],
        "editTask": [Function],
        "editor": Object {},
        "intl": Object {
          "formatMessage": [Function],
        },
        "keyboardShortcutGroups": Object {
          "taskCompletion": Object {
            "cancel": Object {
              "key": "Escape",
              "keyLabel": "ESC",
              "label": "Cancel Editing",
            },
          },
          "taskEditing": Object {
            "editId": Object {
              "key": "e",
              "label": "Edit in Id",
            },
            "editJosm": Object {
              "key": "r",
              "label": "Edit in JOSM",
            },
            "editJosmLayer": Object {
              "key": "t",
              "label": "Edit in new JOSM layer",
            },
            "falsePositive": Object {
              "key": "q",
              "label": "Not an Issue",
            },
            "skip": Object {
              "key": "w",
              "label": "Skip",
            },
          },
        },
        "mapBounds": Object {
          "task": Object {
            "bounds": Array [
              0,
              0,
              0,
              0,
            ],
          },
        },
        "nextTask": [Function],
        "setTaskBeingCompleted": [Function],
        "task": Object {
          "id": 123,
          "status": 0,
        },
        "user": Object {
          "id": 357,
          "settings": Object {
            "defaultEditor": 1,
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
